src/dasi/api/dasi_c.h

```diff
@@ -156,6 +162,27 @@ int dasi_retrieve_next(dasi_retrieve_t* retrieve);
 int dasi_retrieve_attrs(const dasi_retrieve_t* retrieve, dasi_key_t** key,
                         dasi_time_t* timestamp, long* offset, long* length);
 
+/* Wipe functionality */
+
+int dasi_wipe(dasi_t* dasi, const dasi_query_t* query, const dasi_bool_t* doit, const dasi_bool_t* all,
+              dasi_wipe_t** wipe);
+
+int dasi_free_wipe(const dasi_wipe_t* wipe);
+
+int dasi_wipe_next(dasi_wipe_t* wipe);
+
+int dasi_wipe_attrs(const dasi_wipe_t* wipe, const char** value);
```

Comment:
Can we add some doctoring documentation.

Ideally we want this to be for everything (see the relevant file in odc for an example if needed), but here it is especially needed for the value argument. What are the ownership semantics of the return char* value - does it need to be freed (no) or is it pointing to something internal with a restricted validity lifetime (yes).

Note that everywhere else in the API, returning a ** value requires using a paired free function.